pkg/endpoint/api.go

```diff
@@ -338,6 +338,20 @@ func (e *Endpoint) getNamedPortsModel() (np models.NamedPorts) {
 	return np
 }
 
+// getWorkloadModel returns the endpoint's Workload object.
+func (e *Endpoint) getWorkloadModel() *models.Workload {
+	w := e.Workload.Load()
```

Comment:
Is ok if workload is nil:
1. Firstly i comment in `workload` struct definition that this filed may be nil to notice follwers,.
2. Secondly we cannot always get workload, because sometime theres is lack of ownerReference in pod metadata.
3. Thirdly, I set omitempty tag in every filed about workload, so when it comes  to encode/decode, encoder/decoder will handle this correctly!

And thank you for your advice, but due to my mistake(this pr is my master branch, I have to discard all changes to sync with current upstream to handle merging thing), I have to open another pr, sorry for that.

Reply:
Thanks to @gentoo-root , I finally try to get git.
slight ping, Is this ok?